classes/profile_helper.php

```diff
@@ -104,51 +119,34 @@ public static function get_min_duration_for_group_and_reason(string $group, int
      * @return float duration (as seconds) of the fastest profile for a given reason.
      */
     public static function get_min_duration_for_reason(int $reason, bool $usecache = true): float {
-        global $DB;
+        $quota = self::$quotas[$reason];
 
-        $reasonstr = profile::REASON_STR_MAP[$reason];
-        $quota = (int) get_config('tool_excimer', "num_$reasonstr");
-
-        $cachekey = 'profile_type_' . $reason . '_min_duration_s';
-        $result = get_config('tool_excimer', $cachekey);
+        $cachekey = '__all__';
+        $cachefield = 'profile_type_' . $reason . '_min_duration_s';
+        $cache = \cache::make('tool_excimer', 'request_metadata');
+        $result = $cache->get($cachekey);
 
-        if (!$usecache || $result === false) {
+        if (!$usecache || $result === false || !isset($result[$cachefield])) {
             // Get and set cache.
-            $reasons = $DB->sql_bitand('reason', $reason);
+            $db = manager::get_altconnection();
```

Comment:
do we need protection here if no connection like here:

https://github.com/catalyst/moodle-tool_excimer/pull/300/files#diff-6546d87581281c290f904eb79396be0da45288aec26b4159c0a102d85118872fR189

Reply:
I will fix this.